### PR TITLE
Function reinterpretAs updated to support big integers

### LINKED ISSUE #19691: how to convert FixedString(16) to UInt256 in CH?

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Fixes #19691.

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Function `reinterpretAs` updated to support big integers. Fixes #19691.

### CODE REVIEW

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 1]
```diff
@@ -303,81 +303,45 @@ SELECT toFixedString('foo\0bar', 8) AS s, toStringCutToZero(s) AS s_cut
...
-
-This function accepts a number or date or date with time, and returns a FixedString containing bytes representing the corresponding value in host order (little endian). Null bytes are dropped from the end. For example, a UInt32 type value of 255 is a FixedString that is one byte long.
-
-## reinterpretAsUUID {#reinterpretasuuid}
-
-This function accepts 16 bytes string, and returns UUID containing bytes representing the corresponding value in network byte order (big-endian). If the string isn't long enough, the functions work as if the string is padded with the necessary number of null bytes to the end. If the string longer than 16 bytes, the extra bytes at the end are ignored. 
-
-**Syntax**
+Following reinterpretations are allowed:
+1. Any type that has fixed size and value of that type can be represented continuously into FixedString. Null bytes are dropped from the end. For example, a UInt32 type value of 255 is a string that is one byte long.
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/functions/type-conversion-functions.md:311]: For FixedString, dropping null bytes does not make sense.

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 2]
```diff
@@ -303,81 +303,45 @@ SELECT toFixedString('foo\0bar', 8) AS s, toStringCutToZero(s) AS s_cut
...
+2. Any type that if value of that type can be represented continuously into String. Null bytes are dropped from the end. For example, a UInt32 type value of 255 is a string that is one byte long.
+3. Types that can be interpreted as numeric (Integers, Float, Date, DateTime, UUID) into FixedString,
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/functions/type-conversion-functions.md:313]: How is that differ from 1, 2?

File: docs/en/sql-reference/functions/type-conversion-functions.md (hunk 1 above)
Comment (kitaisreal) [reply docs/en/sql-reference/functions/type-conversion-functions.md:311]: Updated.

File: docs/en/sql-reference/functions/type-conversion-functions.md (hunk 2 above)
Comment (kitaisreal) [reply docs/en/sql-reference/functions/type-conversion-functions.md:313]: Updated.

**Pull request merged**